[FX][docs] Document gotcha about training flag

jubinchheda opened the pull request:
Fixes #68913

Commit:
[FX][docs] Document gotcha about training flag #68190

Summary: [FX][docs] Document gotcha about training flag #68190

Test Plan: make html and view differences

Reviewers: jamesr66a

Subscribers: jamesr66a

Tasks:

Tags:

Committer: jubinchheda <jubinchheda@fb.com>

jubinchheda committed:
Update fx.rst

jubinchheda committed:
Merge pull request #1 from jubinchheda/jubinchheda-patch-1

Update fx.rst

Commit:
[FX][docs] Document gotcha about training flag

Summary: [FX][docs] Document gotcha about training flag

Test Plan: make html

Reviewers:  jamesr66a

Subscribers:  jamesr66a

Tasks:

Tags:

Committer: jubinchheda <jubinchheda@fb.com>

Commit:
[FX][docs] Document gotcha about training flag
Summary: [FX][docs] Document gotcha about training flag

Test Plan: make html

Reviewers:  jamesr66a

Subscribers:  jamesr66a

Tasks:

Tags:

Committer: jubinchheda <jubinchheda@fb.com>

Commit:
Summary: [FX][docs] Document gotcha about training flag #68190

Test Plan: make html and view differences

Reviewers: jamesr66a

Subscribers: jamesr66a

Tasks:

Tags:

Committer: jubinchheda jubinchheda@fb.com

Commit:
Summary: [FX][docs] Document gotcha about training flag #68190

Test Plan: make html and view differences

Reviewers: jamesr66a

Subscribers: jamesr66a

Tasks:

Tags:

Committer: jubinchheda jubinchheda@fb.com

jamesr66a commented on docs/source/fx.rst:
```diff
@@ -1006,6 +1006,67 @@ Miscellanea
+
+        x = torch.randn(5, 3)
+        torch.testing.assert_allclose(traced(x), x)
+
```
Maybe add a conclusion sentence like "Because of this difference, you should prefer to mark modules that interact with the `training` flag dynamically as leaf modules"

jamesr66a reviewed (commented):
Looks good! Can you add a screenshot of the final rendered docs section?

Commit:
Summary: [FX][docs] Document gotcha about training flag #68190
Test Plan: make html and view differences

Reviewers: jamesr66a

Subscribers: jamesr66a

Tasks:

Tags:

Committer: jubinchheda jubinchheda@fb.com

Commit:
Summary: [FX][docs] Document gotcha about training flag #68190
Test Plan: make html and view differences

Reviewers: jamesr66a

Subscribers: jamesr66a

Tasks:

Tags:

Committer: jubinchheda jubinchheda@fb.com

jubinchheda commented:
Screenshot of the final rendered docs section:

![Screen Shot 2021-11-29 at 3 19 50 PM](https://user-images.githubusercontent.com/9194904/143937240-c6055ba4-bd54-40e8-97d3-07111a149073.png)
![Screen Shot 2021-11-29 at 3 20 00 PM](https://user-images.githubusercontent.com/9194904/143937238-a15954ab-0f13-4411-8ebf-187fa1c981b2.png)

jamesr66a reviewed (approved):
Looks good, thanks! Please import to FBCode and you can land there
